generators/app/templates/src/containers/TicTacToe/GameListPage/children/List/list.component.js

```diff
@@ -214,31 +214,44 @@ const List = ({ webId, gamePath, sendNotification }: Props) => {
           gameUrls.push(value);
         }
 
+        if (gameUrls.length === 0) {
+          return gameUrls;
+        }
+
         // Run the list of games fetched from the document through the shex validator helper function and store the valid items in gameList
         const gameShapeUrl = 'https://solidsdk.inrupt.net/sdk/tictactoe.shex';
         gameList = await shexUtil.validateList(gameUrls, gameShapeUrl);
 
         let games = [];
-        for await (const item of gameList) {
-          // the url at this point the document is being cached by LDFlex, we need to manually clear it
-          ldflex.clearCache(item);
-          const game = await ldflexHelper.fetchLdflexDocument(item);
-          let gameData = { url: item };
-          if (game) {
-            for await (const field of tictactoeShape.shape) {
-              let values = [];
-              for await (const val of game[getPredicate(field)]) {
-                values = [...values, val.value];
-              }
-              const value = values.length > 1 ? values : values[0];
-              gameData = { ...gameData, [field.predicate]: value };
-            }
-            const opponent = await getPlayerInfo(gameData.opponent);
-            const actor = await getPlayerInfo(gameData.actor);
-            gameData = { ...gameData, opponent, actor, deleted: false, documentUrl: url };
-          } else {
-            gameData = { ...gameData, status: GameStatus.DELETED, deleted: true, documentUrl: url };
+
+        // Looping over all valid game data
+        for await (const game of gameList) {
+          let gameData = {};
+
+          // Looping over each predicate in the game shape
+          for (const field of tictactoeShape.shape) {
+            const predicate = getPredicate(field);
```

Comment:
The function 'getPredicate()' only takes one param and yet knows about `tictactoeShape` internally - shouldn't the Shape be passed to the function instead?

Reply:
Maybe! The goal of the PR was only to update to using the new validateList function, so I left the internal logic of the game fetching alone as much as I could. I can add game list refactoring to the list of tech debt, as additional changes are out of scope for this work item.